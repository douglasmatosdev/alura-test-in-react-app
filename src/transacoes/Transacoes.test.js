import React from 'react'
import { render } from '@testing-library/react'
import Transacoes from './Transacoes'

describe('Componente de transações', () => {
    it('Deve renderizar uma lista de transações', () => {
        const transacoes = [
            { data: '16/12/2020', tipo: 'saque', valor: '50,00' },
            { data: '26/11/2020', tipo: 'deposito', valor: '150,00' },
            { data: '10/10/2020', tipo: 'saque', valor: '100,00' }
        ]
        const { container } = render(
            <Transacoes
                transacoes={transacoes}
            />)

        expect(container.firstChild).toMatchSnapshot()
    })
})