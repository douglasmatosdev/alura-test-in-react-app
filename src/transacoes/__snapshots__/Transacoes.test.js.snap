// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transações Deve renderizar uma lista de transações 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      16/12/2020
    </p>
    <p />
    <p>
      R$ 
      50,00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      26/11/2020
    </p>
    <p />
    <p>
      R$ 
      150,00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      10/10/2020
    </p>
    <p />
    <p>
      R$ 
      100,00
    </p>
  </div>
</div>
`;
